lib/takeaway.rb

```diff
@@ -0,0 +1,81 @@
+require './lib/menu.rb'
+require 'twilio-ruby'
+require './auth_credentials'
+
+class TakeAway
+  attr_reader :basket, :menu, :total
+
+  def initialize(menu: Menu.new)
+    @basket = []
+    @menu = menu
+    @total = 0
+  end
+
+  def read_menu
+    @menu.read_menu
+  end
+
+  def add_to_basket(dessert)
+    raise 'Please pick a dessert from the menu.' unless in_menu?(dessert)
+    adding_to_total(dessert)
+  end
+
+  def remove_from_basket(dessert)
+    @basket.each { |item| item.each { |name, price| @total -= price if name == dessert } }
+    @basket.select { |item| item.each { |name, _| @basket.delete item if name == dessert } }
```

Comment:
the choice here is to delete all items in the array that are that item. Which causes the frustration of not being able to delete 1 at a time.

Either store dish and quantities using a hash, or if you're particularly wedded to the array perhaps use the index of the item to remove it.